source/D2Lang/src/D2Unicode/D2UnicodeStr.cpp

```diff
@@ -35,6 +35,18 @@ Unicode* __fastcall Unicode::strcat(Unicode* dest, const Unicode* src) {
   return dest;
 }
 
+int __fastcall Unicode::strcmp(const Unicode* str1, const Unicode* str2) {
+  for (size_t i = 0; (str1[i].ch != L'\0') || (str2[i].ch != L'\0'); ++i) {
+    if (str1[i].ch < str2[i].ch) {
```

Comment:
Maybe add comment here that we can not overflow since if any of the 2 characters is `L'\0'` then it necessarily enters one of the two branches ?
This is not obvious at a first look and one might think that we can read pas the end of a string

Reply:
I have added a new comment to explain the loop logic.